### DISCUSSION TITLE
How to skip lines, get the header name, and remove tabs

### DISCUSSION THREAD

**Discussion by OpenFoam-User** [johnkerl/miller#1453] [Q&A] [answered]
Hey,

I am a new user, I found about this project only today. I have went through the 10 minutes tutorial, unfortunately, I was disappoined, as it does not contain much information.

I have this file ìnput.txt`: 
[input.txt](https://github.com/johnkerl/miller/files/13753184/input.txt)
 that I am trying to process as follows:

* Skip the first 11 lines
* Use the line `# Time Cd Cl CmPitch` as the header
* Convert it to csv.

Could you please tell me how to proceed?

Thank you

**Comment by aborruso** [comment, marked as answer]
Hi @OpenFoam-User I love the Miller documentation. It's a big project, it takes a little time and practice.

I would do it this way

```
<input.txt tail -n +13 | mlr --t2c clean-whitespace >output.csv
```

Some notes:

- `--t2c` to convert from TSV to CSV
- `clean-whitespace` is a [verb](https://miller.readthedocs.io/en/6.10.0/reference-verbs/index.html#clean-whitespace) to remove any redundant white space,

In this way you get

```
# Time,Cd,Cl,CmPitch
1e-05,5.961839e+03,6.112924e+00,1.461476e+03
2e-05,-2.230266e+02,2.719546e+00,-4.979762e+01
3e-05,-1.804804e+03,1.353850e-02,-4.335161e+02
4e-05,-1.573092e+01,3.288806e+00,-1.137172e+01
5e-05,1.211822e+02,1.414652e+00,3.349247e+01
6e-05,-7.082260e+01,2.963701e+00,-8.245512e+00
7e-05,-2.950213e+01,3.538622e-01,-4.937992e+00
8e-05,-1.459736e+01,3.896183e+00,2.305539e-01
9e-05,-8.436069e+00,5.771371e-01,1.147752e+00
0.0001,-5.127596e+00,4.446924e+00,1.429827e+00
0.00011,-8.304146e+00,5.947566e-01,1.114428e+00
0.00012,-6.788659e+00,4.469475e+00,1.172506e+00
0.00013,-4.741239e+00,8.778420e-01,1.556971e+00
0.00014,-3.521566e+00,4.649601e+00,1.746606e+00
0.00015,-3.079408e+00,9.854932e-01,1.809214e+00
0.00016,-1.939612e+00,4.641109e+00,1.935292e+00
```

**Reply by OpenFoam-User** [reply]
Thank you for your answer :)

**Reply by aborruso** [reply]
@OpenFoam-User  How did you discover Miller today?

**Reply by OpenFoam-User** [reply]
@aborruso :  I asked ChatGPT for a powerful CSV tool 😃 

BTW, can I skip the first lines without using `tail` command? because right now I am using Windows?

**Reply by OpenFoam-User** [reply]
Hi @aborruso : Any updates on this?

Thank you :)

**Reply by aborruso** [reply]
> BTW, can I skip the first lines without using tail command? because right now I am using Windows?

Hi @OpenFoam-User I do not use Miller in windows.

In windows powershell you can run something like

```
Get-Content input.txt | Select-Object -Skip 12
```

to have this

![image](https://github.com/johnkerl/miller/assets/30607/8dc0fe60-c38b-43fa-99c7-aa28c8ddae28)
